src/Microsoft.AspNetCore.Routing/Builder/MapEndpointEndpointDataSourceBuilderExtensions.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Matching;
+using Microsoft.AspNetCore.Routing.Patterns;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    public static class MapEndpointEndpointDataSourceBuilderExtensions
+    {
+        public static EndpointBuilder MapEndpoint(
```

Comment:
Should this return `EndpointBuilder` or `MatcherEndpointBuilder`? There are public properties on MatcherEndpointBuilder now but no one will likely use them because the base type is returned.

Reply:
I will leave it returning `EndpointBuilder` unless someone says otherwise 👼

Reply:
This should be `MatcherEndpointBuilder`. Sorry, I thought I commented to taht effect. Maybe I did it in the wrong spot.